user/configuration-variables.rst

```diff
@@ -0,0 +1,89 @@
+Configuration variables/context in OpenWISP2
+============================================
+
+What configuration variables are, why do we need them?
+------------------------------------------------------
+
+It's often needed for the configuration to differ a little between
+devices, we may need to use device specific values
+(like MAC address, or UUID) in it, in such situation it's not to possible
+to simply use static templates. 
+The configuration variables feature allows us to include in our config
+some default/automatically assigned variables
+and define our own ones in controller admin.
+
+The set of all configuration variables is known as *context*.
+
+Default context
+---------------
+
+Following variables are added automatically
+to the context of each configuration:
+
+- ``id`` - device's UUID
+- ``key`` - device's OpenWISP key
+- ``name`` - device's name in OpenWISP
+- ``mac_address`` - device's MAC address
+
+Referencing variables in configuration
+--------------------------------------
+
+We can include variables in our configuration by simply writing3
+``{{ variable_name }}`` or ``{{variable_name}}`` in it.
+
+Usage example
+~~~~~~~~~~~~~
+
+Using following template (JSON/advanced view notation)
+we can change OpenWISP connection settings on our devices
+without the need to manually set
+``UUID`` and ``key`` parameters in configuration:
+
+.. code-block:: javascript
+
+    {
+      "openwisp": [
+        {
+          "config_name": "controller",
+          "config_value": "http",
+          "url": "http://controller.myopenwisp.org",
+          "interval": "60",
+          "uuid": "{{ id }}",
+          "key": "{{ key }}"
+        }
+      ]
+    }
+
+
+Custom variables
+----------------
+
+We can define our custom variables per device in OpenWISP Controller by
+putting them in *context* field as a JSON object:
+
+.. code-block:: javascript
+
+    {
+      "variable1": "value1",
+      "variable2": "value2",
+      "variable3": "value3",
+      //etc...
+    }
+
+Variables may *only* contain alphanumeric characters and underscores.
+
+Usage example
+~~~~~~~~~~~~~
+
+We may not want our access points to use same SSIDs,
+it's possible to quickly append some text/number to them
+and in the same time leave the opportunity to change the base SSID
+using configuration variables.
+
+Setting the SSID in our template to ``MyWifiNetwork {{ ap_id }}`` and
+``ap_id`` variable to 1, 2, ... for following devices will result
+in having *MyWifiNetwork 1*, *MyWifiNetwork 2*, ... wifi networks.
+
+Note, that undefined variables will **not** be evaluated: not defining
+``ap_id`` variable for some device using template described above
+will result in a wifi network named ``MyWifiNetwork {{ ap_id }}``.
```

Comment:
This is a good example but we should add a full example for it in order to be useful.

First of all, you should create a template which contains the wireless interface with the SSID definition.
As an example, for the SSID let's use something like `"Apartment {{ apartment_id }}"`  or `"School classroom {{ classroom_id }}"`.
Show a screenshot of this.

Then in the device configuration, select the template just created, add the context, eg:

```json
{
    "apartment_id": "B07"
}
```

Add a screenshot which shows the selected template and the context, then add another screenshot which shows the resulting configuration.